code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -1587,36 +1599,47 @@
 	return
 
 // Sliding from one table to another
-/obj/item/weapon/reagent_containers/food/drinks/MouseDropFrom(atom/over_object,atom/src_location,over_location,src_control,over_control,params)
+/obj/item/weapon/reagent_containers/food/drinks/MouseDropFrom(atom/over_object,atom/src_location,atom/over_location,src_control,over_control,params)
 	var/mob/user = usr
 	if (!istype(src_location))
 		return
 	if (!user || user.incapacitated())
 		return
-	if (!user.Adjacent(src) || !src_location.Adjacent(over_location))
-		return
+	// Attempted drink sliding
 	if ((locate(/obj/structure/table) in src_location) && (locate(/obj/structure/table) in over_location))
-		user.visible_message("<span class='notice'>\The [user] slides \the [src] down the table.</span>", "<span class='notice'>You slide \the [src] down the table!</span>")
-		forceMove(over_location, glide_size_override = DELAY2GLIDESIZE(2))
-		return
+		if (M_SOBER in user.mutations)
+			if (!user.Adjacent(src))
+				return
+			var/distance = manhattan_distance(over_location, src)
+			if (distance >= 3) // More than 3 tiles to go
+				return ..()
+		
+			// Geometrically checking if we're on a straight line.
+			var/vector/V = atoms2vector(src, over_location)
+			var/vector/V_norm = V.duplicate()
+			V_norm.noramlize()
+			if (!V_norm.is_integer())
+				return ..() // Only a cardinal vector (north, south, east, west) can pass this test
+			
+			// Checks if there's tables on the path.
+			var/turf/dest = get_translated_turf(V)
+			var/turf/temp_turf = get_translated_turf(V_norm)
+			while (temp_turf != dest)
+				if (!locate(/obj/structure/table) in temp_turf)
+					return ..()
+				temp_turf = temp_turf.get_translated_turf(V_norm)
+			
+			vector_translate(V, 0.2 SECONDS)
+			user.visible_message("<span class='notice'>\The [user] expertly slides \the [src] down the table.</span>", "<span class='notice'>You slide \the [src] down the table. What a pro.</span>")
+			return
+		else
+			if (!user.Adjacent(src) || !src_location.Adjacent(over_location)) // Regular users can only do short slides.
+				return ..()
+			if (M_CLUMSY in user.mutations && prob(10))
```

Comment:
isn't this parsed as `M_CLUMSY in (user.mutations && prob(10))`?

Reply:
Wouldn't surprise me, so I'll adjust that